sdk/src/main/java/com/mapbox/maps/MapboxMap.kt

```diff
@@ -49,11 +49,15 @@ class MapboxMap internal constructor(
   MapCameraManagerDelegate,
   MapStyleStateDelegate {
 
-  internal lateinit var style: Style
+  internal var style: Style? = null
   internal var isStyleLoadInitiated = false
-  private val styleObserver = StyleObserver(this, nativeObserver, pixelRatio)
+  private val styleObserver = StyleObserver(
```

Comment:
Removed `this` param from StyleObserver, using reference to the object not yet constructed fully could lead to subtle bugs. Instead only passed lambda that inits style directly.